includes/class-activitypub.php

```diff
@@ -66,23 +68,24 @@ public static function render_json_template( $template ) {
 		}
 
 		$accept_header = $_SERVER['HTTP_ACCEPT'];
-
-		if (
-			\stristr( $accept_header, 'application/activity+json' ) ||
-			\stristr( $accept_header, 'application/ld+json' )
-		) {
-			return $json_template;
-		}
-
 		// Accept header as an array.
 		$accept = \explode( ',', \trim( $accept_header ) );
-
 		if (
+			\stristr( $accept_header, 'application/activity+json' ) ||
+			\stristr( $accept_header, 'application/ld+json' ) ||
 			\in_array( 'application/ld+json; profile="https://www.w3.org/ns/activitystreams"', $accept, true ) ||
 			\in_array( 'application/activity+json', $accept, true ) ||
 			\in_array( 'application/ld+json', $accept, true ) ||
 			\in_array( 'application/json', $accept, true )
 		) {
+			$secure_mode = \get_option( 'activitypub_use_secure_mode', '0' );
```

Comment:
We have so much issues with caching and content negotiations, that I would not add signature verification on get requests! Besides of that, these requests are synchron and generate a lot of latency for the requester.

Reply:
This is a user setting which could be appreciated by some privacy minded folks. It is off by default.

Reply:
but it uses the same setting that enables/disables the incoming requests. I really would remove this for now and think about that for a furure release!

Reply:
In both cases it only applies to get requests.

Reply:
But following the WordPress philosopgy: Decisions, not options, does it provide any real value, that we need to add it?

(I do not want to convince you both, I want to realy understand why to add it!)

Reply:
There is a clear enough reason not to make it a default: Performance.

But there is also a clear enough reason to keep the option, which fits in with many of WP's own moderation options: Abuse prevention.

> As a result, through the authentication mechanism ... it becomes possible to enforce who can and cannot retrieve even public content from your server, e.g. servers whose domains you have blocked.

Typically WP blogs have been more concerned with SPAM, but [many](https://www.pewresearch.org/internet/2021/01/13/the-state-of-online-harassment/) [studies](https://www.sciencedirect.com/science/article/pii/S8755461521000037) have shown that  harassment is increasingly prevalent in social media. ActivityPub removes even more barriers to interaction with blog posts, so I believe a corresponding option to limit is a fair compromise.

Reply:
OK, agreed. But what about hiding it from the user by using a CONST instead of a setting, to play around a bit for some reseases? Like a feature switch?

And if it works fine, to add a setting afterwards. We could add the documentation about using the CONST in the FAQ?